include/deal.II/lac/arpack_solver.h

```diff
@@ -875,6 +969,23 @@ ArpackSolver::solve(const MatrixType1 & /*system_matrix*/,
 }
 
 
+
+template <typename VectorType, typename MatrixType1>
+inline void
+ArpackSolver::solve(const MatrixType1 &                A,
+                    std::vector<std::complex<double>> &eigenvalues,
+                    std::vector<VectorType> &          eigenvectors,
+                    const unsigned int                 n_eigenvalues)
+{
+  eigenvalue_problem_type                  = 'I';
+  const_cast<Mode &>(additional_data.mode) = regular;
+  MatrixType1 dummy_b;
+  MatrixType1 dummy_inverse;
+  solve(A, dummy_b, dummy_inverse, eigenvalues, eigenvectors, n_eigenvalues);
```

Comment:
I'm not too fond of setting a status variable, creating dummy variables, calling a separate function, and letting that function ignore the dummy arguments because of that status variable. This also creates status for a later time that you probably didn't want to carry over from one `solve()` call to another.

So I think I'm with @davydden: I'd rather you create an (internal) third function that is called from both public `solve()` functions.

Reply:
How is that different? The call to the internal function will be pretty much the same as this call. I will have to create dummy variables that will be ignored in the internal function.

Reply:
As Wolfgang mentioned, the current code is buggy: if you first call solver with standard eigenvalue problem, the subsequent calls for generalized eigensolver wont' do what you think (this flag will have wrong value). To me, this is simply another reason not to adopt this flag mechanism but rather add a private `solve`.

Reply:
Yes, that was my main concern -- state is bad.